File: Projects/leetcode/MinimumEffort.java
package leetcode;

import java.util.Arrays;
import java.util.Comparator;

class MinimumEffort {
    public static int minimumEffort(int[][] tasks) {
        Arrays.sort(tasks, Comparator.comparingInt(task -> task[1] - task[0]));
        int minimumEffort = 0;
        for (int[] task : tasks) {
            minimumEffort = Math.max(minimumEffort + task[0], task[1]);
        }
        return minimumEffort;
    }
}

File: Projects/leetcode/minimumEffort.kt
package leetcode

import java.util.*

fun minimumEffort(tasks: Array<IntArray>): Int {
    Arrays.sort(tasks, Comparator.comparingInt { task: IntArray -> task[1] - task[0] })
    var minimumEffort = 0
    for (task in tasks) {
        minimumEffort = Math.max(minimumEffort + task[0], task[1])
    }
    return minimumEffort
}
